File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int res = 0, curSum = 0;
        unordered_map<int, int> prefixSums;
        prefixSums[0] = 1;

        for (int num : nums) {
            curSum += num;
            int diff = curSum - k;
            res += prefixSums[diff];
            prefixSums[curSum]++;
        }
        //[(0,1)]
        //[(0,1) (2,1)] 1
        //[(0,1) (2,1) (1,1)] 1
        //[(0,1) (2,2) (1,1)] 2
        //[(0,1) (2,2) (1,1) (4,1)] 4

        return res;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] indSum = new int[2];
        for (int i = 0; i < nums.length - 1; i++) {
            for (int k = i + 1; k < nums.length; k++) {
                if (nums[i] + nums[k] == target) {
                    indSum[0] = i;
                    indSum[1] = k;
                }
            }
        }
        return indSum;
    }
}
